apps/charterafrica/src/lib/ecosystem/utils.js

```diff
@@ -0,0 +1,23 @@
+function localize(data, locales = ["en"]) {
+  const getValue = (value, locale) => {
+    if (value?.[locale] !== undefined) {
+      return value?.[locale];
+    }
+    if (value?.en !== undefined) {
+      return value?.en;
+    }
+    return value;
+  };
```

Comment:
If I'm reading this function correctly, you're returning en data if say pt data is missing, correct?

If so, is it the right approach? If `localized` is enabled in Airtable, then it's up to the content/data team to ensure all locales have data, no?

Reply:
Remember this function from `airtable`.

```js
 const locales = localized ? ["en", "fr", "pt"] : ["en"];
  const theme = locales.reduce((curr, acc) => {
    acc[curr] = getValue(data, toolTableColumns.theme[curr])?.[0] ?? null;
    return acc;
  }, {});
  const description = locales.reduce((curr, acc) => {
    acc[curr] = getValue(data, toolTableColumns.description[curr]);
    return acc;
  }, {});
```

means if lacalised is false, object will look like

```js
{
 description: {en: ""},
 theme: {en: ""}
}
```
 so we need to sanitize this as well @kilemensi

Reply:
Yes, but if localized was false there, it will be false here as well @koechkevin

Reply:
@koechkevin

Reply:
@kilemensi If I understood you I have this updated as 

```js
function localize(data, locales = ["en"], localized = false) {
  const getValue = (value, locale) => {
    if (value?.[locale] !== undefined) {
      return value?.[locale];
    }
    if (value?.en !== undefined && !localized) {
      return value?.en;
    }
    return value;
  };

  return locales.reduce((loc, locale) => {
    const localizedData = loc;
    localizedData[locale] = {};
    Object.keys(data).reduce((locData, key) => {
      const localeData = locData;
      localeData[key] = getValue(data[key], locale);
      return localeData;
    }, localizedData[locale]);
    return localizedData;
  }, {});
}
```

Reply:
Why are we returning en value if say pt value is undefined @koechkevin? This is what I'm asking?

The point of `localized` is to make sure we're reading and writing only the data we need. If `localized` is enabled in Airtable, then `fr` and `pt` columns *MUST* contain data. Otherwise we only need `en` data.

Reply:
@kilemensi  We are returning `en` only if localized is `false`

_If localized is enabled in Airtable, then fr and pt columns MUST contain data._ Correct(For only fields that are localisable).